src/test/java/org/openrewrite/java/migrate/MXBeanNonPublicTest.java

```diff
@@ -0,0 +1,304 @@
+
+    @Test
+    void classAndNonPublicInterface() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import javax.management.InstanceAlreadyExistsException;
+              import javax.management.MBeanException;
+              import javax.management.MBeanRegistrationException;
+              import javax.management.MBeanServer;
+              import javax.management.MXBean;
+              import javax.management.NotCompliantMBeanException;
+              import javax.management.ObjectName;
+              import javax.management.ReflectionException;
+              
+              class TestClassAndNonPublicInterface {
+                  public void regMBean(ObjectName objectName, MBeanServer server) throws InstanceAlreadyExistsException, MBeanRegistrationException, NotCompliantMBeanException {
+                      Object obj = new NonPublic();
+                      server.registerMBean(obj, objectName);
+                  }
+              
+                  public void createMBean(ObjectName objectName, MBeanServer server) throws InstanceAlreadyExistsException, NotCompliantMBeanException, ReflectionException, MBeanException {
+                      server.createMBean("bean", objectName);
+                  }
+              
+                  class NonPublic implements NonPublicMBean {
+                  }
+              
+                  @MXBean
+                  private interface NonPublicMBean {
+                  }
+              }
+              """,
+            """
+              import javax.management.InstanceAlreadyExistsException;
+              import javax.management.MBeanException;
+              import javax.management.MBeanRegistrationException;
+              import javax.management.MBeanServer;
+              import javax.management.MXBean;
+              import javax.management.NotCompliantMBeanException;
+              import javax.management.ObjectName;
+              import javax.management.ReflectionException;
+                             
+              class TestClassAndNonPublicInterface {
+                  public void regMBean(ObjectName objectName, MBeanServer server) throws InstanceAlreadyExistsException, MBeanRegistrationException, NotCompliantMBeanException {
+                      Object obj = new NonPublic();
+                      server.registerMBean(obj, objectName);
+                  }
+              
+                  public void createMBean(ObjectName objectName, MBeanServer server) throws InstanceAlreadyExistsException, NotCompliantMBeanException, ReflectionException, MBeanException {
+                      server.createMBean("bean", objectName);
+                  }
+              
+                  class NonPublic implements NonPublicMBean {
+                  }
+              
+                  @MXBean
+                  public interface NonPublicMBean {
+                  }
+              }
+              """
+          )
+        );
+    }
```

Comment:
What are we really testing here? seems overly complex for the one line we aim to change here. I'm inclined to fully remove this test.
```suggestion
```

Reply:
This test captures the sub class scenario. The test was a port of our test from our eclipse plugin. The test likely captures scenarios that our other migration rules depended on. I could trim it down if you want.

Reply:
Trimming it down would help; now it's a bit unclear what elements are and aren't necessary for what we're testing. Minimal tests are also easier to step through with the debugger if there's no excess classes, methods or invocations.

Reply:
No problem. I'll have that update up soon.